docs/msiv1_token_revocation.md

```diff
@@ -0,0 +1,48 @@
+# MSAL Support for MSI v1 Token Revocation and Capabilities
+
+This specification describes **two optional parameters** that MSAL can send to the **MSI v1 endpoint** (e.g., `http://169.254.169.254/metadata/identity/oauth2/token`) when acquiring tokens via **Managed Identity**. These parameters empower developers to control **token revocation** and specify **client capabilities** that might alter the token issuance behavior by Azure AD (ESTS).
+
+---
+
+## Overview
+
+When MSAL requests a token from the MSI v1 endpoint for resources like `https://management.azure.com/`, it typically reuses valid, locally cached tokens. However, certain scenarios require explicitly **ignoring** the cache or specifying **client capabilities** that Azure AD uses to enable or disable particular features.
```

Comment:
I am confused by this spec. `bypass_cache` and `xms_cc` are new internal params for MSIv1. We don't expose these to app developers. What do we expose to app developers ? WithClaims and WithClientCapabilities?

If so, you need the APIs from the app developers to the ppl with Claims. 

We should also have details about how to E2E test this (even if manual) including how to invoke token revocation.

Reply:
> What do we expose to app developers ? WithClaims and WithClientCapabilities?

[`WithClaims` has already been exposed](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/4.68.0/src/client/Microsoft.Identity.Client/ApiConfig/AcquireTokenForManagedIdentityParameterBuilder.cs#L74).  `WithClientCapabilities` seems unnecessary because CAE is automatically enabled for S2S traffic, possibly also for MSI.

As to bypass_cache and xms_cc, they are internal for sure. And it seems no service is currently supporting them. We can follow up the conversation on Teams.

Reply:
addressed comments

Reply:
@rayluo we do have a way to test this now. I've added the doc link